src/EditorFeatures/Core/Implementation/IntelliSense/DocumentProvider.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense
+{
+    internal class DocumentProvider : IDocumentProvider
```

Comment:
there's one impl of this, and we use it in what appears to be two locations.  i have no idea why we created this, but i'm ok with you just bruning it to the ground and inlining GetDocument.

Alternatively, add an extension to ITextSnapshot called GetDocumentWithFrozenPartialSemantics and update all callsites to use that instead.

Reply:
works for me

Reply:
➡️ Leaving this for a follow-up due to the complexity of the tests involved.